src/components/PopupWithForm.js

```diff
@@ -37,4 +54,9 @@ export default class PopupWithForm extends Popup {
   _resetForm() {
     this._popupForm.reset();
   }
+
+  close() {
+    super.close();
+    this._resetForm();
```

Comment:
You need to **reset** the form only in block `then` not to do that immediately after `submit` because when there is an error the popup will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more

You should **clear ** the inputs only after **submitting** a card rather than when you open or close popups because if you unexpectedly close the popup you will be upset if the inputs are empty when you open it again and you’ll have to type the data again.